### PR TITLE
Globalfixes next138

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
This patchset is composed of a number of changes that affect the whole project in one step. This includes:
- Namespace changes for TPM types
- Adding of const qualifiers to SAPI functions
- Changing the return code of almost all functions

### CODE REVIEW

File: test/unit/TPML-marshal.c [hunk 1]
```diff
@@ -43,20 +42,18 @@ tpml_marshal_success(void **state)
     sel.pcrSelections[1].pcrSelect[0] = 0xdd;
     sel.pcrSelections[1].pcrSelect[1] = 0xee;
 
-    ptr2 = (TPML_PCR_SELECTION *)(buffer);
-
     rc = Tss2_MU_TPML_PCR_SELECTION_Marshal(&sel, buffer, buffer_size, NULL);
-    assert_int_equal (rc, TSS2_RC_SUCCESS);
```
Comment (tstruk) [inline test/unit/TPML-marshal.c]: We are skipping the check of rc after marshaling.

File: test/unit/TPML-marshal.c [hunk 2]
```diff
@@ -43,20 +42,18 @@ tpml_marshal_success(void **state)
...
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[1], 0xbb);
-    assert_int_equal (ptr2->pcrSelections[0].pcrSelect[2], 0xcc);
-    assert_int_equal (ptr2->pcrSelections[1].hash, HOST_TO_BE_16(TPM_ALG_SHA256));
-    assert_int_equal (ptr2->pcrSelections[1].sizeofSelect, 2);
-    assert_int_equal (ptr2->pcrSelections[1].pcrSelect[0], 0xdd);
-    assert_int_equal (ptr2->pcrSelections[1].pcrSelect[1], 0xee);
+
+    uint8_t expect [] = {
+        0, 0, 0, 2,
+        (uint8_t)(TPM_ALG_SHA1 / 256), (uint8_t)TPM_ALG_SHA1,
```
Comment (tstruk) [inline test/unit/TPML-marshal.c]: This is quite confusing. (uint8_t)0x0004 / 256 will be 0 anyway. Why not just 0, 0x04?

File: test/unit/TPML-marshal.c [hunk 3]
```diff
@@ -43,20 +42,18 @@ tpml_marshal_success(void **state)
...
+        3,
+        0xaa, 0xbb, 0xcc,
+        (uint8_t)(TPM_ALG_SHA256 / 256), (uint8_t)TPM_ALG_SHA256,
```
Comment (tstruk) [inline test/unit/TPML-marshal.c]: Same here: Why not just 0, 0x0b?

File: test/unit/TPML-marshal.c (hunk 1 above)
Comment (AndreasFuchsSIT) [reply test/unit/TPML-marshal.c]: Ups, that was our bad... Remove one too many lines...
Will update...

File: test/unit/TPML-marshal.c (hunk 2 above)
Comment (AndreasFuchsSIT) [reply test/unit/TPML-marshal.c]: We wanted to make it explicit what is written into these... I guess it is a matter of style... We could replace it with
`0, 0x04, /* TPM_ALG_SHA1 */`
Would you prefer this ?

**Comment by AndreasFuchsSIT** [comment]
Sent updated pull-request... See #619

File: test/unit/TPML-marshal.c (hunk 2 above)
Comment (tstruk) [reply test/unit/TPML-marshal.c]: It was just the div by 256. 0, 0x04, /* TPM_ALG_SHA1 */ looks good to me.

**Pull request closed without merging**